teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -107,20 +108,60 @@ export class MapsService {
     return await this.mapsRepository.findOne({ where: { id: mapId }} );
   }
 
-  getDeletedAt(lastModified: Date, afterDays: number): Date {
+  async getDeletedAt(map: MmpMap, afterDays: number): Promise<Date> {
+    // get newest node of this map:
+    const newestNodeQuery = this.nodesRepository
+                                .createQueryBuilder()
+                                .select("max(MmpNode.lastModified) AS lastModified")
+                                .where( { nodeMapId: map.id } )
+    const newestNode = newestNodeQuery.getRawOne()
+    const newestNodeLastModified = (await newestNode)["lastmodified"]
+    const lastModified = (newestNodeLastModified === null) ? map.lastModified : newestNodeLastModified;
+    
+    return this.calculcateDeletedAt(new Date(lastModified), afterDays)
+  }
+
+  calculcateDeletedAt(lastModified: Date, afterDays: number): Date {
     // dont modify original input as this might be used somewhere else
     const copyDate: Date = new Date(lastModified.getTime());
     copyDate.setDate(copyDate.getDate() + afterDays);
     return copyDate;
   }
 
-  async deleteOutdatedMaps(afterDays: number = 30): Promise<DeleteResult> {
-    return this.mapsRepository
+  async deleteOutdatedMaps(afterDays: number = 30): Promise<Number> {
+    const today = new Date();
+
+    const deleteQuery = this.mapsRepository
       .createQueryBuilder()
-      .where("(lastModified + (INTERVAL '1 day' * :afterDays)) < :today", { afterDays: afterDays, today: new Date() })
-      .delete()
-      .from(MmpMap)
-      .execute();
+      .select("MmpMap.id")
+      .leftJoin(qb => 
+        // subquery to get the newest node and its lastModified date of this map:
+        qb
+        .select(["node.nodeMapId AS nodeMapId", "max(node.lastModified) AS lastUpdatedAt"])
+        .from("MmpNode", "node")
+        .groupBy("node.nodeMapId"), "lastmodifiednode", "lastmodifiednode.nodeMapid = MmpMap.id")
+      .where(
+        // delete all maps that have nodes that were last updated after afterDays
+        "(lastmodifiednode.lastUpdatedAt + (INTERVAL '1 day' * :afterDays)) < :today", { afterDays: afterDays, today: today})
```

Comment:
`{ afterDays: afterDays, today: today}` can be written as `{ afterDays, today }`